src/application/jobs/maj-mailing-list-conseiller.job.handler.ts

```diff
@@ -47,51 +42,61 @@ export class MajMailingListConseillerJobHandler extends JobHandler<Job> {
       resultat: {}
     }
 
-    const contactsMilo =
-      await this.mailRepository.findAllContactsConseillerByStructures([
-        Core.Structure.MILO
-      ])
-    const contactsPoleEmploi =
-      await this.mailRepository.findAllContactsConseillerByStructures([
-        Core.Structure.POLE_EMPLOI
-      ])
-    const contactsBRSA =
-      await this.mailRepository.findAllContactsConseillerByStructures([
-        Core.Structure.POLE_EMPLOI_BRSA
-      ])
-    const contactsAIJ =
-      await this.mailRepository.findAllContactsConseillerByStructures([
-        Core.Structure.POLE_EMPLOI_AIJ
-      ])
-    const contactsCD =
-      await this.mailRepository.findAllContactsConseillerByStructures([
-        Core.Structure.CONSEIL_DEPT
-      ])
-    await this.mailService.mettreAJourMailingList(
-      contactsMilo,
-      parseInt(this.mailingLists.milo)
-    )
-    await this.mailService.mettreAJourMailingList(
-      contactsPoleEmploi,
-      parseInt(this.mailingLists.poleEmploi)
-    )
-    await this.mailService.mettreAJourMailingList(
-      contactsBRSA,
-      parseInt(this.mailingLists.brsa)
-    )
-    await this.mailService.mettreAJourMailingList(
-      contactsCD,
-      parseInt(this.mailingLists.cd)
-    )
-    const conseillersSansEmail =
-      await this.mailRepository.countContactsConseillerSansEmail()
-    const stats = {
-      conseillersMilo: contactsMilo.length,
-      conseillersPoleEmploi: contactsPoleEmploi.length,
-      conseillersBRSA: contactsBRSA.length,
-      conseillersAIJ: contactsAIJ.length,
-      conseillersSansEmail
+    const mailingLists = {
```

Comment:
On peut le typer en `Map<Core.Structure, { id: string }>` ou `{ [key in Core.Structure]: { id: string }` pour avoir une erreur si on implémente pas toutes les structures.

Reply:
ok